### DISCUSSION TITLE
Memory usage issue

### DISCUSSION THREAD

**Discussion by shiftie** [vercel/next.js#42475] [Help]
Hello,

I'm observing surprising memory usage when generating pages.
I'm using next 12.2.5 (tried with 12.3.2, same outcome), I've tried `isrMemoryCacheSize: 0,` but the memory still goes up while generating pages & never actually goes down.

My test consists to set `revalidate` to 1 second, then request 1000 different pages.
After test is done, I repeat with same URLs (that are rebuilt since revalidation delay has expired).

Here is the chart of memory used:
![Screenshot 2022-11-04 at 09 44 43](https://user-images.githubusercontent.com/1401223/199932735-68246575-f3c8-492e-85d9-e29040c1af23.jpg)
note: there are 4 lines since the app runs in 4 tasks

As you can see, memory always goes up.

I doubt this is normal since that means that memory usage is growing with the amount of pages being generated (which can be in millions if the server doesn't reboot for a given time) & the app will eventually go out of memory and die?

Do you have an idea why this is happening?

**Comment by icyJoseph** [comment]
This is really interesting. I had seen some people claim memory usage increase, but never had a repository where this could be reliably done.

- Does this still happen upgrading to a more recent version?
- Do images change between ISR rounds?
- Do you have any 3rd party libraries that could be holding onto a reference that blocks the garbage collector?

Is it possible to see this repository? is it a shareable experiment?

**Reply by shiftie** [reply]
Hi!
So I've tried with a minimal nextjs app with ISG. 
On local, i generate 1000 pages with ISG & I output `process.memoryUsage().rss`.
Before test it's like 70mb, during test it spikes to like 120mb.
Then after 15-20s, it's back to around 70mb.
I run same image & same test on ECS Fargate, there it spikes but it doesn't go down the same, so it seems to be relative to that environment.
